src/query/service/src/sessions/query_ctx.rs

```diff
@@ -761,12 +761,17 @@ impl TableContext for QueryContext {
 
     fn add_segment_location(&self, segment_loc: Location) -> Result<()> {
         let mut segment_locations = self.inserted_segment_locs.write();
-        segment_locations.push(segment_loc);
+        segment_locations.insert(segment_loc);
         Ok(())
     }
 
     fn get_segment_locations(&self) -> Result<Vec<Location>> {
-        Ok(self.inserted_segment_locs.read().to_vec())
+        Ok(self
+            .inserted_segment_locs
+            .read()
+            .iter()
+            .cloned()
+            .collect::<Vec<_>>())
```

Comment:
Does the order of `inserted_segment_locs` make sense?

Reply:
Nop, order is insensitive